packages/ui/src/components/ProjectCard/ProjectCard.tsx

```diff
@@ -1,75 +1,55 @@
 import React from 'react';
-import { Box, Flex, forwardRef, Grid } from '@chakra-ui/react';
-
-import { statusValueCheck } from '../../helpers/statusValueCheck';
-import { Button } from '../Button/Button';
-import { Typography } from '../Typography/Typography';
-
-type DoneProps = {
-  status: 'done';
-  points: number;
-  pointsMax: number;
-  date?: never;
-};
-
-type NotDoneProps = {
-  status: 'idle' | 'doing' | 'review';
-  date: Date;
-  points?: never;
-  pointsMax?: never;
-};
-
-type CommonProps = {
-  image: string;
-  title: string;
-  url: string;
-};
-
-export type ProjectCardProps = CommonProps & (DoneProps | NotDoneProps);
-
-export const ProjectCard = forwardRef<ProjectCardProps, 'div'>(
-  // disabled for url props
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  ({ status, image, title, url, points, pointsMax, date, ...props }, ref) => {
-    const { statusText, timeStatus, dateOrPoints, isDoing } = statusValueCheck(
-      status,
-      points as number,
-      pointsMax as number,
-      date as Date,
-    );
-
-    return (
-      <Box status={status} ref={ref} w="100%" h="100%" boxShadow="base" borderRadius="8px" {...props}>
-        <Grid templateRows="repeat(2, 1fr)" h="100%" gap="24px">
-          <Box as="img" src={image} alt="" background="gray.300" borderRadius="8px 8px 0 0" h="auto" />
-
-          <Grid templateRows="repeat(3 1fr)" justifyContent="center">
-            <Typography as="p" fontWeight="700" size="lg" lineHeight="xl" letterSpacing="4xl">
-              {title}
+import { useHistory } from 'react-router-dom';
+import { Box, Flex, Grid, Image } from '@chakra-ui/react';
+
+import { Button } from '../Button';
+import { Typography } from '../Typography';
+import { DoneProps, NotDoneProps, ProjectCardProps } from './projectCardTypes';
+import { statusValueCheck } from './statusValueCheck';
+
+export const ProjectCard = ({ status, image, title, url, points, pointsMax, date, ...props }: ProjectCardProps) => {
+  const history = useHistory();
```

Comment:
🤔 ui components shouldn't rely on any specific routing library. We want to use them both in the panel and website (where we have Next.js). You can add `onButtonClick` prop to the component, so we can customize the logic of the function when implementing this component